src/BinSkim.Rules/PERules/BA2101.EnableIntegrityCheck.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Composition;
+using System.Reflection.PortableExecutable;
+
+using Microsoft.CodeAnalysis.BinaryParsers;
+using Microsoft.CodeAnalysis.BinaryParsers.PortableExecutable;
+using Microsoft.CodeAnalysis.IL.Sdk;
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Driver;
+
+namespace Microsoft.CodeAnalysis.IL.Rules
+{
+    [Export(typeof(Skimmer<BinaryAnalyzerContext>)), Export(typeof(ReportingDescriptor))]
+    public class EnableIntegrityCheck : PEBinarySkimmerBase
+    {
+        /// <summary>
+        /// BA2101
+        /// </summary>
+        public override string Id => RuleIds.EnableIntegrityCheck;
+
+        /// <summary>
+        /// Binaries that are loaded by certain Windows features must (and device drivers should) 
+        /// opt into Windows validation of their digital signatures by setting the /INTEGRITYCHECK 
+        /// linker flag. This option sets the IMAGE_DLLCHARACTERISTICS_FORCE_INTEGRITY attribute 
+        /// in the PE header of binaries which tells the memory manager to validate a binary's 
+        /// digital signature when loaded. Any user mode code that is interfacing with Early Launch 
+        /// Antimalware (ELAM) drivers, integrates with device firmware execution or is trying to 
+        /// load into protected process lite space must enable /INTEGRITYCHECK. This feature applies 
+        /// to both 32-but and 64-bit files. Binaries that opt into /INTEGRITYCHECK must be signed
+        /// using the Microsoft Azure Code Signing program.
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString
+        {
+            Text = RuleResources.BA2101_EnableIntegrityCheck_Description
+        };
+
+        public override bool EnabledByDefault => false;
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+            nameof(RuleResources.BA2101_Pass),
+            nameof(RuleResources.BA2101_Error),
+            nameof(RuleResources.NotApplicable_InvalidMetadata)
+        };
+
+        public const uint IMAGE_DLLCHARACTERISTICS_FORCE_INTEGRITY = 0x0080;
```

Comment:
>IMAGE_DLLCHARACTERISTICS_FORCE_INTEGRITY [](http://example.com/codeflow?start=26&length=40)

Comment the source code, this flag doesn't ship yet in .net

Reply:
[https://learn.microsoft.com/en-us/dotnet/api/system.reflection.portableexecutable.dllcharacteristics?view=net-7.0](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.portableexecutable.dllcharacteristics?view=net-7.0)